web-common/src/features/dashboards/time-series/MetricsTimeSeriesCharts.svelte

```diff
@@ -182,25 +183,12 @@
     $timeControlsStore.selectedTimeRange &&
     !isScrubbing
   ) {
-    if (!mouseoverValue?.x || !(mouseoverValue.x instanceof Date)) {
-      chartInteractionColumn.update((state) => ({
-        ...state,
-        hover: undefined,
-      }));
-    } else {
-      const { position: columnNum } = bisectData(
-        mouseoverValue.x,
-        "center",
-        "ts_position",
-        isAllTime ? formattedData?.slice(1) : formattedData?.slice(1, -1),
-      );
-
-      if ($chartInteractionColumn?.hover !== columnNum)
-        chartInteractionColumn.update((state) => ({
-          ...state,
-          hover: columnNum,
-        }));
-    }
+    updateChartInteractionStore(
+      mouseoverValue?.x,
+      undefined,
+      isAllTime,
+      formattedData,
+    );
```

Comment:
**Performance Issue**: The updateChartInteractionStore method is called within a reactive statement which can lead to excessive calls and potential performance bottlenecks if the dependencies change frequently. <br> **Fix**: Use a debounced or throttled version of updateChartInteractionStore to reduce the frequency of updates, especially when dependent on high-frequency reactive variables. <br> **Code Suggestion**: 
 ```
 +  import { throttle } from 'lodash';
 +  const throttledUpdate = throttle(updateChartInteractionStore, 200);
 +  throttledUpdate(
 +    mouseoverValue?.x,
 +    undefined,
 +    isAllTime,
 +    formattedData,
 +  );
 ```